PROTO_tests/tests/dataframe_test.py

```diff
@@ -1167,12 +1296,441 @@ def test_to_markdown(self):
+        assert df.columns.values==good_labels
+        assert df["one1"][0]==1
+        assert df["three3"][0]=="foo"
+        assert df["four4"][1]==-1.8
+
+        with pytest.raises(ValueError):
+            df = ak.DataFrame(unlabeled_data, columns=bad_labels1)
+        with pytest.raises(ValueError):
+            df = ak.DataFrame(unlabeled_data, columns=bad_labels2)
+        with pytest.raises(TypeError):
+            df = ak.DataFrame(unlabeled_data, columns=["one", "two", 3, "four"])
+
+    def test_from_pandas(self):
+        username = ["Alice", "Bob", "Alice", "Carol", "Bob", "Alice", "John", "Carol"]
+        userid = [111, 222, 111, 333, 222, 111, 444, 333]
+        item = [0, 0, 1, 1, 2, 0, 0, 2]
+        day = [5, 5, 6, 5, 6, 6, 1, 2]
+        amount = [0.5, 0.6, 1.1, 1.2, 4.3, 0.6, 0.5, 5.1]
+        bi = 2**200
+        bi_arr = [bi, bi + 1, bi + 2, bi + 3, bi + 4, bi + 5, bi + 6, bi + 7]
+        ref_df = pd.DataFrame(
+            {
+                "userName": username,
+                "userID": userid,
+                "item": item,
+                "day": day,
+                "amount": amount,
+                "bi": bi_arr,
+            }
+        )
+
+        df = ak.DataFrame(ref_df)
+        assert ((ref_df == df.to_pandas()).all()).all()
+
+        df = ak.DataFrame.from_pandas(ref_df)
+        assert ((ref_df == df.to_pandas()).all()).all()
+
+    def test_to_pandas(self):
+        df = self.build_ak_df()
+        pd_df = self.build_pd_df()
+
+        assert_frame_equal(pd_df,df.to_pandas())
+
+        slice_df = df[ak.array([1, 3, 5])]
+        pd_df = slice_df.to_pandas(retain_index=True)
+        assert pd_df.index.tolist()==[1,3,5]
+
+        pd_df = slice_df.to_pandas()
+        assert pd_df.index.tolist()==[0,1,2]
+
+    def test_merge(self):
+        df1 = ak.DataFrame(
+            {
+                "key": ak.arange(4),
+                "value1": ak.array(["A", "B", "C", "D"]),
+                "value3": ak.arange(4, dtype=ak.int64),
+            }
+        )
+
+        df2 = ak.DataFrame(
+            {
+                "key": ak.arange(2, 6, 1),
+                "value1": ak.array(["A", "B", "D", "F"]),
+                "value2": ak.array(["apple", "banana", "cherry", "date"]),
+                "value3": ak.ones(4, dtype=ak.int64),
+            }
         )
-        assert df.to_markdown(tablefmt="jira") == df.to_pandas().to_markdown(tablefmt="jira")
 
+        ij_expected_df = ak.DataFrame(
+            {
+                "key": ak.array([2, 3]),
+                "value1_x": ak.array(["C", "D"]),
+                "value3_x": ak.array([2, 3]),
+                "value1_y": ak.array(["A", "B"]),
+                "value2": ak.array(["apple", "banana"]),
+                "value3_y": ak.array([1, 1]),
+            }
+        )
+
+        ij_merged_df = ak.merge(df1, df2, how="inner", on="key")
+
+        assert ij_expected_df.columns.values == ij_merged_df.columns.values
```

Comment:
Can we replace all these asserts with `assert_frame_equal(ij_expected.to_pandas(retain_index=True), ij_merged.to_pandas(retain_index=True)` to simplify these tests a bit?